### PR TITLE
Clean up of std.uni

### PR DESCRIPTION
**Pull request by JackStouffer**
Separated into three disparate commits:

* remove analogs to `std.math.nextPow2` and `std.math.trucPow2` to keep DRY
* remove custom analog of `std.algorithm.comparison.equal`
* fix doc strings on functions and make several undocumented public functions package

### CODE REVIEW

File: std/uni.d [hunk 1]
```diff
@@ -1706,7 +1695,7 @@ unittest
 
 
 // Simple storage manipulation policy
-@trusted public struct GcPolicy
+@trusted package struct GcPolicy
```
Comment (wilzbach) [inline std/uni.d]: Is this used somewsomewhere else except for this module?

Comment (DmitryOlshansky) [inline std/uni.d]: Nope its internal, keep it private

File: std/uni.d [hunk 2]
```diff
@@ -3978,7 +3978,7 @@ public:
     }
 }
 
-/*
+/**
```
Comment (DmitryOlshansky) [inline std/uni.d]: It never went public

Comment (JackStouffer) [inline std/uni.d]: > It never went public

It's been public since 2013

**Comment by JackStouffer** [comment]
@wilzbach fixed

File: std/uni.d (hunk 2 above)
Comment (DmitryOlshansky) [inline std/uni.d]: Okay let me rephrase it - it was never _intended_ to be public. The whole thing was too specialized to be of use to wide audience.

**Comment by JackStouffer** [comment]
@DmitryOlshansky fixed.

**Comment by DmitryOlshansky** [comment]
> @DmitryOlshansky fixed.

buildTrie is still in the docs, it should not be.

**Comment by JackStouffer** [comment]
@DmitryOlshansky Made it private.

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Comment by JackStouffer** [comment]
Thanks!
